url542ae48576_tonyheadford_m68k_tgz-pJ8-m68k_MonitorJ8/src/m68k/cpu/CpuCore.java

```diff
@@ -0,0 +1,2574 @@
+		{
+			return false;
+		}
+
+		public int getTiming()
+		{
+			return size == Size.Long ? 10 : 6;
+		}
+
+		// used for jmp and jsr, lea, movem
+		public int getComputedAddress()
+		{
+			return address;
+		}
+
+		public int index()
+		{
+			return index;
+		}
+
+		public String toString()
+		{
+			return new StringBuilder(5).append("-(a").append(regNumber).append(")").toString();
+		}
+	}
+
+	class AddressRegisterWithDisplacementOperand implements Operand
+	{
+		protected int regNumber;
+		protected Size size;
+		protected int address;
+		protected int displacement;
+		protected final int index = 5;
+
+		public void init(int param, Size sz)
+		{
+			regNumber = param;
+			size = sz;
+			displacement = fetchPCWordSigned();
+			address = getAddrRegisterLong(regNumber) + displacement;
+		}
+
+		public int getByte()
+		{
+			return readMemoryByte(address);
+		}
+
+		public int getWord()
+		{
+			return readMemoryWord(address);
+		}
+
+		public int getLong()
+		{
+			return readMemoryLong(address);
+		}
+
+		public int getByteSigned()
+		{
+			return readMemoryByteSigned(address);
+		}
+
+		public int getWordSigned()
+		{
+			return readMemoryWordSigned(address);
+		}
+
+		public void setByte(int value)
+		{
+			writeMemoryByte(address, value);
+		}
+
+		public void setWord(int value)
+		{
+			writeMemoryWord(address, value);
+		}
+
+		public void setLong(int value)
+		{
+			writeMemoryLong(address, value);
+		}
+
+		public boolean isRegisterMode()
+		{
+			return false;
+		}
+
+		public boolean isSR()
+		{
+			return false;
+		}
+
+		public int getTiming()
+		{
+			return size == Size.Long ? 12 : 8;
+		}
+
+		// used for jmp and jsr
+		public int getComputedAddress()
+		{
+			return address;
+		}
+
+		public int index()
+		{
+			return index;
+		}
+
+		public String toString()
+		{
+			return new StringBuilder(10).append(String.format("$%x",displacement)).append("(a").append(regNumber).append(")").toString();
+		}
+	}
+
+	class AddressRegisterWithIndexOperand implements Operand
+	{
+		protected int regNumber;
+		protected Size size;
+		protected int address;
+		protected int displacement;
+		protected int idxRegNumber;
+		protected Size idxSize;
+		protected boolean idxIsAddressReg;
+		protected final int index = 6;
+
+		public void init(int param, Size sz)
+		{
+			regNumber = param;
+			size = sz;
+			int ext = fetchPCWordSigned();
+			displacement = signExtendByte(ext);
+			idxRegNumber = (ext >> 12) & 0x07;
+			idxSize = ((ext & 0x0800) == 0x0800 ? Size.Long : Size.Word);
+			idxIsAddressReg = ((ext & 0x8000) == 0x8000);
+			int idxVal;
+			if(idxIsAddressReg)
+			{
+				if(idxSize == Size.Word)
+				{
+					idxVal = getAddrRegisterWordSigned(idxRegNumber);
+				}
+				else
+				{
+					idxVal = getAddrRegisterLong(idxRegNumber);
+				}
+			}
+			else
+			{
+				if(idxSize == Size.Word)
+				{
+					idxVal = getDataRegisterWordSigned(idxRegNumber);
+				}
+				else
+				{
+					idxVal = getDataRegisterLong(idxRegNumber);
+				}
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 1869:1890, 2259:2280.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.